kernel/aster-nix/src/net/socket/unix/stream/socket.rs

```diff
@@ -273,6 +277,36 @@ impl Socket for UnixStreamSocket {
 
         connected.write(buf)
     }
+
+    fn sendmsg(&self, msg_hdr: MessageHeader, flags: SendRecvFlags) -> Result<usize> {
+        let MessageHeader {
+            addr,
+            io_vecs,
+            control_message,
+        } = msg_hdr;
+
+        let packet = copy_packet_from_user(&io_vecs);
+        let sent_len = self.sendto(&packet, addr, flags)?;
+
+        Ok(sent_len)
+    }
+
+    fn recvmsg(&self, dst: Box<[IoVec]>, flags: SendRecvFlags) -> Result<(usize, MessageHeader)> {
+        debug_assert!(flags.is_all_supported());
```

Comment:
I have found a number of consistency issues between different implementations of `sendmsg/recvmsg`, including but not limited to the following
 - `debug_assert!(flags.is_all_supported());`
 - `warn!("receiving control message is not supported");`
 - not really important, but how you put empty lines between function calls really varies a lot.

For the current implementation, I propose to unify all these implementations into two methods, regardless of whether the socket is TCP/UDP/Unix. How would you like to just make them the default implementations of `Socket`?

In fact, I think we will eventually do the reverse: make `sendto()`/`recvto()` call `sendmsg()`/`recvmsg()` by default, since the functionality of the latter is a superset of the former (and I think Linux already does this, though I have not checked the source). I think it's okay to do it in this PR, how do you feel about it?

Reply:
> In fact, I think we will eventually do the reverse: make sendto()/recvto() call sendmsg()/recvmsg() by default, since the functionality of the latter is a superset of the former (and I think Linux already does this, though I have not checked the source). I think it's okay to do it in this PR, how do you feel about it?

Sure, the latter makes more sense. `sendto()`/`recvto()` should call `sendmsg()`/`recvmsg()` by default. I will refactor the code accordingly.